samples/java-protobuf-doc-snippets/src/it/java/com/example/JwtIntegrationTest.java

```diff
@@ -0,0 +1,68 @@
+/* This code was generated by Kalix tooling.
+ * As long as this file exists it will not be re-generated.
+ * You are free to make changes to this file.
+ */
+package com.example;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import kalix.javasdk.Metadata;
+import kalix.javasdk.testkit.junit.KalixTestKitResource;
+import org.junit.ClassRule;
+import org.junit.Test;
+
+import java.util.Base64;
+import java.util.Map;
+import java.util.concurrent.CompletionStage;
+
+import static java.util.concurrent.TimeUnit.SECONDS;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+
+// Example of an integration test calling our service via the Kalix proxy
+// Run all test classes ending with "IntegrationTest" using `mvn verify -Pit`
+public class JwtIntegrationTest {
+
+  /**
+   * The test kit starts both the service container and the Kalix proxy.
+   */
+  @ClassRule
+  public static final KalixTestKitResource testKit =
+      new KalixTestKitResource(Main.createKalix());
+
+  /**
+   * Use the generated gRPC client to call the service through the Kalix proxy.
+   */
+  // tag::bearer-token-claims-test[]
+  private final JwtServiceActionClient client;
+
+  public JwtIntegrationTest() {
+    client = (JwtServiceActionClient) testKit.getGrpcClient(CounterService.class); // <1>
+  }
+
+  @Test
+  public void testMsgWithClaim() throws Exception {
+    String bearerToken = bearerTokenWith( // <2>
+        Map.of("iss", "my-issuer", "sub", "hello"));
+
+    var msg = "Hello from integration test";
+    var req = JwtService.MyRequest.newBuilder().setMsg(msg).build();
+    var response = client.jwtInToken()
+        .addHeader("Authorization", "Bearer " + bearerToken) // <3>
+        .invoke(req)
+        .toCompletableFuture()
+        .get();
+
+    assertTrue(response.getMsg().contains(msg));
+  }
+
+  private String bearerTokenWith(Map<String, String> claims) throws JsonProcessingException {
+    // setting algorithm to none
+    String alg = "ewogICJhbGciOiAibm9uZSIKfQo"; // <4>
+    byte[] jsonClaims = new ObjectMapper().writeValueAsBytes(claims);
+
+    // no validation is done for integration tests, thus no valid signature required
+    return alg + "." +  Base64.getEncoder().encodeToString(jsonClaims); // <5>
+  }
```

Comment:
we could eventually build some small test utility to provide this part out of the box.. but I'm not so convinced it's worth the increase in tesktit API surface. I think this example in docs might be enough. LKM if disagree.